### PR TITLE
STITCH-2499 Add support for 'watch' in React Native

### PR DESCRIPTION
**Pull request by adamchel**
This adds support for "watch" functionality in React Native, as requested by https://github.com/mongodb/stitch-js-sdk/issues/209.

However, this change depends on https://github.com/facebook/react-native/pull/25718 getting accepted and released in a future release of React Native:  

I will be keeping this PR open until that happens.

### CODE REVIEW

**Comment by BrianIto** [comment]
Hello everybody! I'm just a Junior Developer and just want to understand what's going on right now. React Native added support for the proper networking components to work?

**Comment by adamchel** [comment]
This won't be supported until React Native 0.62 is released.

File: packages/react-native/core/src/core/internal/net/RNFetchStreamTransport.ts [hunk 1]
```diff
@@ -51,6 +53,37 @@ import {
...
+        headers: reqHeaders,
+        method: request.method,
+        mode: 'cors'
+      }).then(response => {
+        const respHeaders: { [key: string]: string } = {};
+        response.headers.forEach((value, key) => {
+          respHeaders[key] = value;
+        });
+        if (response.status < 200 || response.status >= 300) {
+          return response.text()
```
Comment (tkaye407) [inline packages/react-native/core/src/core/internal/net/RNFetchStreamTransport.ts]: Does response.text() return a promise?

File: packages/react-native/services/mongodb-remote/__tests__/RemoteMongoClientIntTests.ts [hunk 2]
```diff
@@ -749,4 +753,262 @@ describe("RemoteMongoClient", () => {
 
     expect(coll.findOne()).rejects.toBeTruthy();
   });
+
+   /* tslint:disable:no-string-literal */
+   it("should properly support watch streams on set of ids", async () => {
```
Comment (tkaye407) [inline packages/react-native/services/mongodb-remote/__tests__/RemoteMongoClientIntTests.ts]: I think you should split this up into smaller tests personally. This is essentially just 4 tests put into one test

Comment (adamchel) [reply packages/react-native/services/mongodb-remote/__tests__/RemoteMongoClientIntTests.ts]: Discussed offline, these tests are basically just copy-pasted from the `watch` tests in the other two SDKs. We're going to leave as is.

File: packages/react-native/core/src/core/internal/net/RNFetchStreamTransport.ts (hunk 1 above)
Comment (adamchel) [reply packages/react-native/core/src/core/internal/net/RNFetchStreamTransport.ts]: yup

Comment (adamchel) [reply packages/react-native/core/src/core/internal/net/RNFetchStreamTransport.ts]: from `lib.dom.d.ts`

```ts
interface Body {
    readonly body: ReadableStream<Uint8Array> | null;
    readonly bodyUsed: boolean;
    arrayBuffer(): Promise<ArrayBuffer>;
    blob(): Promise<Blob>;
    formData(): Promise<FormData>;
    json(): Promise<any>;
    text(): Promise<string>;
}
```

**Pull request merged**